### DISCUSSION TITLE
What's the purpose of making the "combinations.json" file?

### DISCUSSION THREAD

**Discussion by pzhlkj6612** [miurahr/aqtinstall#747] [Q&A]
Hi.

After answering a lot of questions about "Specified Qt version is unknown", I feel dumb: why did aqt tell users it doesn't know a version of Qt? The actual version info is on the remote server and aqt always checks it before downloading anything.

I didn't carefully check all code of aqt, though.

It could be nice if anyone could point me out, thank you in advance!

**Comment by ddalcino** [comment]
We would need a response from @miurahr to know for sure, but IIRC it was meant for parameter validation.

Theoretically, we could use the MetadataFactory for parameter validation instead; doing it that way would make it easier to write more useful error messages, i.e. "that's not a valid architecture, please choose from these 3 options instead," etc.

Btw, good question.

**Comment by miurahr** [comment]
I originally design `aqt` without `list-qt` command, then `combinations.json` was crafted to validate parameters in version 0.x and 1.x days.
As @ddalcino said, we can now use `MetadataFactory` and deprecate it in future.
